docs/rules/camelcase.md

```diff
@@ -107,6 +109,48 @@ var obj = {
 };
 ```
 
+### ignoreDestructuring: false
+
+Examples of **incorrect** code for this rule with the default `{ "ignoreDestructuring": false }` option:
+
+```js
+/*eslint camelcase: "error"*/
+
+var { category_id } = query;
+
+var { category_id = 1 } = query;
+
+var { category_id: category_id } = query;
+
+var { category_id: category_alias } = query;
+
+var { category_id: categoryId, ...other_props } = query;
+```
+
+### ignoreDestructuring: true
+
+Examples of **correct** code for this rule with the `{ "ignoreDestructuring": true }` option:
+
+```js
+/*eslint camelcase: ["error", {ignoreDestructuring: true}]*/
+
+var { category_id } = query;
+
+var { category_id = 1 } = query;
+
+var { category_id: category_id } = query;
+```
+
+Examples of **incorrect** code for this rule with the `{ "ignoreDestructuring": true }` option:
```

Comment:
I would love to see this moved to being first in the section, but that's not a blocker for merging this. (Other rules tend to do incorrect than correct in most sections, but this is probably okay as is.)

Reply:
Sounds good, will push that.